VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -32,101 +33,129 @@
+
+            Intent resultIntent = new Intent(context, CourseDetailTabActivity.class);
+            resultIntent.setAction(Long.toString(System.currentTimeMillis()));
+            //if user launch the app from recent list, activity will still get this intent.
+            //this is one way to avoid it.
+            resultIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
+            resultIntent.putExtra(Router.EXTRA_BUNDLE, courseBundle);
+
+            // Because clicking the notification opens a new ("special") activity, there's
+            // no need to create an artificial back stack.
+            context.startActivity(resultIntent);
+        }
     }
 
-    protected java.lang.Class<? extends android.app.Activity> getActivity(android.content.Context context, android.content.Intent intent) {
+    protected java.lang.Class<? extends android.app.Activity> getActivity(Context context, Intent intent) {
         CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
         return ParseHandleHelper.getActivityClass(payload);
     }
 
-    protected android.app.Notification getNotification(android.content.Context context, android.content.Intent intent) {
+    protected android.app.Notification getNotification(Context context, Intent intent) {
         try {
-
             CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
             Class activity = ParseHandleHelper.getActivityClass(payload);
-            if( activity == CourseDetailTabActivity.class ){
-                String titleTemplate =  ResourceUtil.getResourceString(R.string.COURSE_ANNOUNCEMENT_NOTIFICATION_TITLE);
+            if (activity == CourseDetailTabActivity.class) {
+                String titleTemplate = ResourceUtil.getResourceString(R.string.COURSE_ANNOUNCEMENT_NOTIFICATION_TITLE);
 
                 EventBus.getDefault().postSticky(new CourseAnnouncementEvent(
                         CourseAnnouncementEvent.EventType.MESSAGE_RECEIVED, payload.getCourseId()));
 
-                Bundle courseBundle = new Bundle();
-                courseBundle.putBoolean(Router.EXTRA_ANNOUNCEMENTS, true);
-                courseBundle.putString(Router.EXTRA_COURSE_ID, payload.getCourseId());
-
-
-                Intent resultIntent = new Intent(context, CourseDetailTabActivity.class);
-                resultIntent.setAction(Long.toString(System.currentTimeMillis()));
-                //if user launch the app from recent list, activity will still get this intent.
-                //this is one way to avoid it.
-                resultIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK |Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
-                resultIntent.putExtra( Router.EXTRA_BUNDLE, courseBundle);
-
-                // Because clicking the notification opens a new ("special") activity, there's
-                // no need to create an artificial back stack.
```

Comment:
That would require some refactoring in the activities that would be in the back-stack. Such a change is out of scope of this fix, but I left the comment for later.